File: 0049-group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hashmap = {}
        for string in strs:
            key = "".join(sorted(string))
            hashmap[key] = hashmap.get(key, []) + [string]
        return hashmap.values()

File: 0049-group-anagrams/solution.go
func groupAnagrams(strs []string) [][]string {
    store := make(map[[26]int][]string)
    for _, str := range strs {
        hash := [26]int{}
        for _, c := range str {
            hash[c - 'a']++
        }
        if strings, exists := store[hash]; exists {
            store[hash] = append(strings, str)
        } else {
            store[hash] = []string{str}
        }
    }
    ans := [][]string{}
    for _, value := range store {
        ans = append(ans, value)
    }
    return ans
}
